### PR TITLE
perfect-hash: init at 0.4.1

### PR DESCRIPTION
**Pull request by KAction**
Description:

    Generate a minimal perfect hash function for a given set of keys. A
    given code template is filled with parameters, such that the output
    is code which implements the hash function. Templates can easily be
    constructed for any programming language.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
     /nix/store/inldg49pn8a6d2y44sx12hmyvjf6hh7z-perfect-hash-0.4.1 139826760
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/perfect-hash/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  # Archive on pypi does not contain examples, which are very helpful to
+  # understand how to use this program, so we use git source.
+  src = fetchFromGitHub {
+    owner = "ilanschnell";
+    repo = "perfect-hash";
+    rev = version;
+    sha256 = "0gkc3n613hl0q4jknrh2nm1n96j97p36q9jjgarb9d8yii9q7792";
+  };
+  postInstall = ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/perfect-hash/default.nix:14]:
```suggestion
  };

  postInstall = ''
```

File: pkgs/development/tools/misc/perfect-hash/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    longDescription = ''
+      Generate a minimal perfect hash function for a given set of keys.
+      A given code template is filled with parameters, such that the
+      output is code which implements the hash function. Templates can
+      easily be constructed for any programming language.
+    '';
+
+    license = licenses.bsd3;
+    maintainers = [ maintainers.kaction ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/perfect-hash/default.nix:32]:
```suggestion
    longDescription = ''
      Generate a minimal perfect hash function for a given set of keys.
      A given code template is filled with parameters, such that the
      output is code which implements the hash function. Templates can
      easily be constructed for any programming language.
    '';
    license = licenses.bsd3;
    maintainers = [ maintainers.kaction ];

```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 97823` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>perfect-hash</li>
  </ul>
</details>

**Comment by KAction** [comment]
@SuperSandro2000 Applied.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 97823` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>perfect-hash</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 97823` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>perfect-hash</li>
  </ul>
</details>

**Pull request merged**